src/vs/platform/terminal/common/xterm/shellIntegrationAddon.ts

```diff
@@ -238,6 +238,13 @@ export class ShellIntegrationAddon extends Disposable implements IShellIntegrati
 						this._createOrGetCommandDetection(this._terminal).setIsWindowsPty(value === 'True' ? true : false);
 						return true;
 					}
+					case 'Task': {
+						const commandDetection = this.capabilities.get(TerminalCapability.CommandDetection);
+						if (commandDetection) {
+							commandDetection.disableCommandStorage();
+							this._logService.debug('#disableCommandStorage');
```

Comment:
Let's leave this log out as it shouldn't be needed when debugging prod, could also use `commandDetection?.` then too